DeepSpeech.py

```diff
@@ -10,37 +9,41 @@
 os.environ['TF_CPP_MIN_LOG_LEVEL'] = sys.argv[log_level_index] if log_level_index > 0 and log_level_index < len(sys.argv) else '3'
 
 import datetime
-import json
-import numpy as np
+import pickle
 import shutil
 import subprocess
-import tempfile
 import tensorflow as tf
 import time
-import importlib
-import cgi
-import pickle
 
-from collections import OrderedDict
-from math import ceil, floor
+from six.moves import zip, range, filter, urllib, BaseHTTPServer
 from tensorflow.contrib.session_bundle import exporter
 from tensorflow.python.tools import freeze_graph
+from threading import Thread, Lock
+from util.data_set_helpers import SwitchableDataSet, read_data_sets
 from util.gpu import get_available_gpus
-from util.spell import correction
 from util.shared_lib import check_cupti
+from util.spell import correction
 from util.text import sparse_tensor_value_to_texts, wer
-from util.data_set_helpers import SwitchableDataSet
 from util.website import maybe_publish
 from xdg import BaseDirectory as xdg
-from six.moves import zip, range, filter, urllib, BaseHTTPServer
-from threading import Thread, Lock
+
+# Hack: tf.app.flags only supports nargs='?', so we have to add our own method
+def DEFINE_list(flag_name, default_value, docstring):
+    tf.app.flags._global_parser.add_argument('--' + flag_name,
+                                             nargs='+',
+                                             help=docstring,
+                                             default=default_value,
+                                             type=str)
```

Comment:
`os.pathsep` varies by OS so our bin/run-* scripts would require extra complexity to use the correct syntax, so I'm gonna go with a comma separated list for now.